### PR TITLE
Feature/web hook amqp

### PR DESCRIPTION
**Pull request by mahlunar**
Currently, when running in multiple replicas, pods can have stale data due to a Github web hook only being received in one of the instances. 

This change adds AMQP  fanout so that Github web hooks are forwarded to each replica to mitigate staleness.

### CODE REVIEW

**Comment by mahlunar** [comment]
I am unsure how to do a proper end to end test. I think that is needed here to avoid breaking flows

File: go.mod [hunk 1]
```diff
@@ -32,7 +32,10 @@ require (
 	k8s.io/client-go v0.24.3
 )
 
-require github.com/gorilla/mux v1.8.0
+require (
+	github.com/gorilla/mux v1.8.0
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+)
```
Comment (Crevil) [inline go.mod:38]: Is this change required or something from testing?

File: internal/broker/amqpextra/amqpextra.go [hunk 2]
```diff
@@ -46,6 +48,9 @@ func New(c Config) (*Worker, error) {
 		return nil, err
 	}
 
+	Exchange = c.Exchange
+	ExchangeFanout = fmt.Sprintf("%s-git-fanout", c.Exchange)
```
Comment (Crevil) [inline internal/broker/amqpextra/amqpextra.go:52]: Should this be `-github-fanout` just for explicitness?

File: internal/broker/amqpextra/consumer.go [hunk 3]
```diff
@@ -15,27 +18,48 @@ import (
...
+			Exchange:            ExchangeFanout,
+			DurableExchange:     false,
+			AutoDeletedExchange: true,
+			Queue:               fmt.Sprintf("%s-git-fanout-%s", w.config.Queue, fanoutID),
+			ExclusiveQueue:      true,
+			DurableQueue:        false,
+			Prefetch:            0,
+			RoutingPatterns:     []string{"#"},
+			Handle: func(message *amqp.Delivery) error {
+				return nil
```
Comment (Crevil) [inline internal/broker/amqpextra/consumer.go:55]: Shouldn't we call the fanout handlers here somewhere?

**Review by Crevil — COMMENTED** [review]
Overall I think this is great. As discussed I think we should see if we can get a test in here to verify the functionality.

File: go.mod (hunk 1 above)
Comment (mahlunar) [reply go.mod:35]: `go mod tidy` did something strange here. Both dependencies were there before. I have merged them back in to the original list.

**Comment by mahlunar** [comment]
@Crevil I figured out how to test this locally and can see I have a connection from the fanout exchange to the local queue.

Please review again and let's discuss how we could automate tests as well.

**Review by Crevil — APPROVED** [review]
